src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSWebWorker.cs

```diff
@@ -51,7 +51,7 @@ public static async Task RunAsync(Func<Task> body, CancellationToken cancellatio
         private static Task<T> RunAsyncImpl<T>(Func<Task<T>> body, CancellationToken cancellationToken)
         {
             var parentContext = SynchronizationContext.Current ?? new SynchronizationContext();
-            var tcs = new TaskCompletionSource<T>();
+            var tcs = new TaskCompletionSource<T>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
it might be nice to put a comment on each of these describing (if anything) what issue it fixes

Reply:
This didn't break any tests we have right now, but I hope it would mean that continuation would not be running synchronously in the `JSWebWorker` because we are about to kill it.

Reply:
added comment